src/Validation/Validator.php

```diff
@@ -1941,6 +1929,42 @@ public function regex($field, $regex, $message = null, $when = null)
         ]);
     }
 
+    /**
+     * Gets the required message for a field
+     *
+     * @param string $field Field name
+     * @return string
+     */
+    public function getRequiredMessage($field)
+    {
+        $defaultMessage = 'This field is required';
+        if ($this->_useI18n) {
+            $defaultMessage = __d('cake', 'This field is required');
+        }
+
+        return isset($this->_presenceMessages[$field])
+            ? $this->_presenceMessages[$field]
+            : $defaultMessage;
+    }
+
+    /**
+     * Gets the empty message for a field
+     *
+     * @param string $field Field name
+     * @return string
+     */
+    public function getEmptyMessage($field)
```

Comment:
getRequired vs getEmpty seems an API contradiction.
You seem to want to name it `getNotEmptyMessage`.
Also, didnt we go from empty to blank for the reason of confusion in PHP core code usage and behavior of empty()?

Reply:
> Also, didnt we go from empty to blank for the reason of confusion in PHP core code usage and behavior of empty()?

No, we changed `Validation::notEmpty()` to `Validation::notBlank()` because of confusion between the former and `Validator::notEmpty()`.

Reply:
You're right about the naming, it should be `getNotEmptyMessage`. As for notBlank, it doesn't handle custom not empty messages the same way notEmpty does. We would have to iterate through all the rules and find the notBlank one to extract the message, share the `_allowEmptyMessages` var, or create a new var on validator to explicitly store notBlank messages. Pick your poison (detecting the rule seems the best to me).